tests/spread/extensions/env-injector/task.yaml

```diff
@@ -0,0 +1,68 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  cd "$SNAP_DIR"
+  snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the hello command is present
+  [ -f squashfs-root/usr/local/bin/hello ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous
+
+  # Create envfile
+  echo 'HELLO_WORLD=Hello World' >> envfile.env
+
+  # Set env vars: Global
+  snap set env-injector-hello env.global="World"
+
+  # Set env vars: specific to each app
+  snap set env-injector-hello apps.hello1.env.hello="Hello"
+  snap set env-injector-hello apps.hello2.env.specific="City"
+
+  # To Check if key with dot will be rejected 
+  snap set env-injector-hello env.word.dot="wrong"
+
+  # To test order of env vars
+  echo 'ORDER="From envfile"' >> envfile.env
+  snap set env-injector-hello apps.hello1.env.order="from app-specific"
+  snap set env-injector-hello envfile="${SNAP_DIR}"/envfile.env
+
+  # Run the hello command
+  # Those will fail if the env vars are not set as expected
+  env-injector-hello.hello1
+  env-injector-hello.hello2
+
+  # Test if the alias it's rewrite
```

Comment:
```suggestion
  # Make sure that alias is NOT rewritten
```

Reply:
How is this checking that the alias isn't rewritten? The set application alias is same as the app name, so there is no way to know if it was rewritten. The alias should be different, as that's the whole point of allowing user-defined aliases.

Then the test can look into squashfs-root/meta/snap.yaml to check the environment before starting configurations. Alternatively, it can continue configuring it and then run it with an expectation to fail (when alias is rewritten).

Reply:
Fixed. Thanks